### ISSUE TITLE
fetch module: failed to transfer an existing file

### ISSUE THREAD

**Issue by sparkvilla** [ansible/ansible#19287]
##### ISSUE TYPE
 - Bug Report
 

##### COMPONENT NAME

fetch

##### ANSIBLE VERSION

```
ansible 2.2.0.0


```

##### OS / ENVIRONMENT

  hosts file:
  [nas]
  ds ansible_host=172.20.0.5 ansible_user=admin 

  ansible.cfg file:
  [ssh_connection]
  scp_if_ssh=True
  [defaults]
  hostfile = ./hosts
  retry_files_enabled=False #do not create retry files
  stdout_callback = json

##### SUMMARY

I have ubuntu 16.04 as a control machine and Synology as target host. I would like to fetch an existing file  `test_ans` from the synology at  `/var/services/homes/admin/test_ans`  and copy it over to the control machine at  `/tmp/`. Ansible sees the file, i.e. the stat module in the playbook returns `stat_result.stat.exists==true`  but is not able to fetch it.   

##### STEPS TO REPRODUCE

ansible-playbook -i hosts keys.yml -e "target=ds"

```
---
- hosts: '{{ target }}'
  tasks:
  - name: Check admin pub keys are present on synology
    stat:
      path: /var/services/homes/admin/test_ans
    register: stat_result
   - name: Downloading pub key
    fetch:
      src: /var/services/homes/admin/test_ans
      dest: /tmp/
      flat: yes
```

##### ACTUAL RESULTS

```
"msg": "failed to transfer file to /tmp/test_ans: /var/services/homes/admin/test_ans: No such file or directory\n"

```

**Comment by Schnitzel** [comment]
had the same issue, but most probably fixed with https://github.com/ansible/ansible/pull/18673

**Comment by sparkvilla** [comment]
@Schnitzel Are u saying that this is not more an issue in Ansible v2.3.0.0? thanks

**Comment by Schnitzel** [comment]
@sparkvilla
I'm using the tag `v2.2.1.0-0.2.rc2` right now and it is no issues anymore, so I believe in `v2.2.1.0` this will be already fixed.

**Comment by billyfoss** [comment]
As someone who ran into this and does not have 2.2.1.0 available via normal channels yet, I found the synchronize module in pull mode to be a workable substitute.  It does require pulling the full directory (or adding rsync include parameters), but it allowed me to finish the role I was working on.

**Comment by Shrews** [comment]
Given that we have not seen similar issues since this was reported, and the versions mentioned here are quite old, it's likely safe to assume that the issue was fixed with #18673, so I'm going to go ahead and close this. If it is discovered that this is still and issue, feel free to open a new issue.
